src/cluster.cpp

```diff
@@ -93,6 +93,22 @@ auto cluster::latency_aware_routing(
     return false;
 }
 
+auto cluster::speculative_execution(
+    std::chrono::milliseconds delay,
+    int                       max_executions) -> bool
+{
+    if (m_cass_cluster_ptr != nullptr)
+    {
+        CassError error = cass_cluster_set_constant_speculative_execution_policy(
+            m_cass_cluster_ptr.get(),
+            static_cast<cass_uint64_t>(delay.count()),
```

Comment:
std::chrono::milliseconds can be negative I think, might want to check and return false if invalid

Reply:
Added a greater than zero check